test.py

```diff
@@ -362,13 +362,12 @@ def create_server(cluster_name: str, seeds: List[str], config_from_test: dict[st
                 cmdline_options=cmdline_options,
                 config_options=config_options)
 
-            # Suite artifacts are removed when
-            # the entire suite ends successfully.
-            self.artifacts.add_suite_artifact(self, server.stop_artifact)
+            self.artifacts.add_exit_artifact(self, server.stop_artifact)
```

Comment:
The old code was fine. suite artifacts are not guaranteed to be called if the harness throws an exception. We should not leak servers under any circumstances. It's fine to invoke stop twice, it's not possibly fine to invoke uninstall without first invoking stop. If you want a change like that, please make sure that uninstall will always call stop first for all kinds of servers (and clusters).

Reply:
Brought the suite artifact back.